app/forms/prior_authority/steps/primary_quote_form.rb

```diff
@@ -82,6 +83,41 @@ def save_quote
         record.update!(attributes.except('service_type', 'custom_service_name', 'file_upload')
                                  .merge(default_attributes))
       end
+
+      def reset_quote_cost_fields
+        return unless service_type_changed?
+
+        values_to_reset = {}
+
+        # If we have changed cost type, e.g. from per_item to per_hour, or we have changed item type,
+        # e.g. from pages to words, then values previously entered shouldn't carry across
+        values_to_reset.merge!(QUOTE_ITEM_FIELD_RESETS) if previous_service_rule.item != current_service_rule.item
+
+        # Cost type is per_item, per_hour, or variable
+        values_to_reset.merge!(QUOTE_ITEM_FIELD_RESETS) if current_service_rule.cost_type != :per_item
```

Comment:
I don't think we want to start filling in values that the user has not chosen. The steer from Mary was to lean towards making the user re-enter more information if they change the service type rather than 'try to make sense' of the data we already have. I was on the fence about what to do if we change from per_x to variable. I think we want to ensure that they see that they now have a choice of cost types and make them confirm which one they want, not just assume based on what they entered when they had no choice in the matter. But if I previously filled in item fields for an enforced per_item service, then I change to a variable service, if I then actively select per_item, should my previously entered values be waiting for me in the conditional reveal? My gut is no, but I could be convinced otherwise.